homeassistant/components/octoprint.py

```diff
@@ -29,8 +33,11 @@ def setup(hass, config):
     """Set up the OctoPrint component."""
     base_url = 'http://{}/api/'.format(config[DOMAIN][CONF_HOST])
     api_key = config[DOMAIN][CONF_API_KEY]
+    number_of_tools = config[DOMAIN][CONF_NUMBER_OF_TOOLS]
+    bed = config[DOMAIN][CONF_BED]
 
-    hass.data[DOMAIN] = {"api": None}
+    hass.data[DOMAIN] = {"api": None, CONF_NUMBER_OF_TOOLS: number_of_tools,
```

Comment:
It is not very pretty to put the config into `hass.data`. You have a API class, so extend this and use only `hass.data[xy] = api`